gapic/src/main/com/google/gapid/models/Follower.java

```diff
@@ -231,6 +234,21 @@ public void onFollow(Path.Any path) {
       case STATE:
         listeners.fire().onStateFollowed(path);
         break;
+      case RESOURCE_DATA:
+        onFollowResource(path.getResourceData());
+        break;
+      default:
+        LOG.log(WARNING, "Unknown follow path result: " + path);
+    }
+  }
+
+  private void onFollowResource(Path.ResourceData path) {
+    Service.Resource r = resources.getResource(path).resource;
```

Comment:
`resources.getResource(path)` can return `null`. We should check it here. I think for now it's OK to simply log it.

Reply:
This is still de-referencing a null. It needs to be:
```
Resources.Resource res = resources.getResource(path);
if (res != null) {
  switch (res.resource.getType()) {
    ...
  }
}
```